x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -554,10 +517,24 @@ private static void analyzeAndMaybeRetry(
         logicalPlanListener.onResponse(plan);
     }
 
-    static Set<String> fieldNames(LogicalPlan parsed, Set<String> enrichPolicyMatchFields) {
+    private static void resolveFieldNames(LogicalPlan parsed, EnrichResolution enrichResolution, ActionListener<ListenerResult> l) {
+        try {
+            // we need the match_fields names from enrich policies and THEN, with an updated list of fields, we call field_caps API
+            var enrichMatchFields = enrichResolution.resolvedEnrichPolicies()
+                .stream()
+                .map(ResolvedEnrichPolicy::matchField)
+                .collect(Collectors.toSet());
+            // get the field names from the parsed plan combined with the ENRICH match fields from the ENRICH policy
+            l.onResponse(fieldNames(parsed, enrichMatchFields, new ListenerResult(enrichResolution)));
+        } catch (Exception ex) {
+            l.onFailure(ex);
+        }
```

Comment:
Why do we need the try-catch clause now? It wasn't present before. I guess that was a bug, where we didn't pass on exceptions from resolution to the listener? Is there a way to reproduce this issue?